[dataviz] feat: add graph Active contributors grouped by age

NatNgs opened the pull request:
**related issues** #1726

---

### Add user-growth plot in Tournesol's streamlit app

Added a new plot in Streamlit, displaying the number of contributors, new users and active users in one line plot.

"Active" users are defined as having done at least one comparison on the week or before, and having done at least one comparison on the week or after.

![Preview](https://github.com/tournesol-app/tournesol/assets/10955538/d19ee992-3751-43db-af44-1e161ed466ad)

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [x] The tests pass and have been updated if relevant
- [ ] The code quality check pass

❤️ Thank you for your contribution!

[development-guidelines]: https://github.com/tournesol-app/tournesol/blob/main/CONTRIBUTING.md#development-guidelines

GresilleSiffle reviewed (commented):
To define with the rest of the team:

What is an active contributor?

NatNgs commented:
Updated graph after comments on discord; updated graph now look like this:

![image](https://github.com/tournesol-app/tournesol/assets/10955538/d2bc74b7-1015-44c7-88f7-cb970c111968)

- Number of categories (=colors, see legend) can be modified easily as needed
- Choice of colors (as the rainbow gradient here) to be confirmed
- Also not sure about how I labelled it ("Age of community", "First comparison date = last comparison date")

GresilleSiffle commented:
Hi @NatNgs 

The new graph is really cool :ok_hand: 

I ran `black -l 99` on the file to make it more readable, and updated some comments and texts to make them more explicit:
- the title is now "Active contributors grouped by age", to introduce the notion of active contributors;
- there is an info box explaining what is an active contributor to avoid any confusion;
- the label of the Y axis is now "Active contributors" to match the title.

The colours and the groups look good to me for now. Let's check with the rest of the team.

 > Also not sure about how I labelled it ("Age of community", "First comparison date = last comparison date")

I haven't updated  "First comparison date = last comparison date" yet, but I'll think about it.

GresilleSiffle commented on data-visualization/pages/03_Community_evolution.py:
```diff
@@ -8,10 +8,16 @@
 import pandas as pd
+import plotly.express as px
 import streamlit as st
+from dateutil.relativedelta import relativedelta
```
This is not a direct dependency of the project, should we add it to the file `requirements.txt` ?

GresilleSiffle commented on data-visualization/pages/03_Community_evolution.py:
```diff
@@ -82,7 +88,106 @@ def add_comparisons_evolution():
+        drop=True
+    )  # Keep only the required data, remove duplicates.
+
+    df.week_date = pd.to_datetime(df.week_date, infer_datetime_format=True, utc=True).astype(
```
Using `astype` raises a warning in the console, and will raise an exception in the future.

> Using .astype to convert from timezone-aware dtype to timezone-naive dtype is deprecated and will raise in a future version.  Use obj.tz_localize(None) or obj.tz_convert('UTC').tz_localize(None) instead

I don't know the best practice here (I have nearly zero experience with pandas), but it looks like the `astype` is used to make methods such as `df.week_date.min()` and `.max()` work. It should be possible to keep the timezone aware datetime, right? Maybe by explicitly sorting the column `week_date`?

If there is no side effect we may want to explicitly convert the timezone aware datetime to naive dates. 

In case it helps:

```python
# works
df.week_date = pd.to_datetime(df.week_date, infer_datetime_format=True, utc=True)
df.week_date.cat.as_ordered().max()

# doesn't work
df.week_date = pd.to_datetime(df.week_date, infer_datetime_format=True, utc=True)
df.week_date.max()
*** TypeError: Categorical is not ordered for operation max
you can use .as_ordered() to change the Categorical to an ordered one
```

lfaucon reviewed (approved):
Thanks @NatNgs
